src/realm/sync/client.hpp

```diff
@@ -484,42 +484,27 @@ class Session {
         /// identity and access rights of the current user.
         std::string signed_user_token;
 
-        /// ClientReset is used for both async open and client reset. If
-        /// client_reset is not util::none, the sync client will perform
-        /// async open for this session if the local Realm does not exist, and
-        /// client reset if the local Realm exists. If client_reset is
-        /// util::none, an ordinary sync session will take place.
+        /// The presence of the ClientReset config indicates an ongoing or
+        /// requested client reset operation. If client_reset is util::none or
+        /// if the local Realm does not exist, an ordinary sync session will
+        /// take place.
         ///
-        /// A session will perform async open by downloading a state Realm, and
-        /// some metadata, from the server, patching up the metadata part of
-        /// the Realm and finally move the downloaded Realm into the path of
-        /// the local Realm. After completion of async open, the application
-        /// can open and use the Realm.
-        ///
-        /// A session will perform client reset by downloading a state Realm, and
-        /// some metadata, from the server. After download, the state Realm will
-        /// be integrated into the local Realm in a write transaction. The
-        /// application is free to use the local realm during the entire client
-        /// reset. Like a DOWNLOAD message, the application will not be able
-        /// to perform a write transaction at the same time as the sync client
+        /// A session will perform client reset by downloading a fresh copy of
+        /// the Realm from the server at a different file path location. After
+        /// download, the fresh Realm will be integrated into the local
+        /// Realm in a write transaction. The application is free to use the
```

Comment:
Updated the comment to be more specific that reading and writing are both allowed. Also noted that if the application writes to the local Realm before the client reset operation has obtained the write lock, the application changes may be lost because they will be counted the same as other local writes made while offline. But this depends on the client reset mode, and if we end up implementing automatic recovery they would be recovered as well.